horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/Util.java

```diff
@@ -844,13 +848,24 @@ public static CheckResult castCheck(String input, String target, EntityManager e
                         results.get(0)[2] == null ? "" : results.get(0)[2].toString());
     }
 
+    public static DecomposedJsonPath decomposeJsonPath(String jsonpath) {
```

Comment:
I do not think we have users using jsonpath filtering. My concern is that if we are going to define a subset of jsonpaths that we can support then we should also have a well defined response when the input is a valid jsonpath but does not meet our requirements

Reply:
I just checked in the current prod implementation and something like `$..name` is not actually supported as it won't pass the `Util.castCheck` call, precisely here https://github.com/Hyperfoil/Horreum/blob/6127bb10560cae96dd3ea4c23a1a04d4aba874b6/horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/TestServiceImpl.java#L707-L722 (it will go in else-else case) ending up in returning all label values completely ignoring the provided filter! 

In my implementation, it will throw an error which I think is much better than ignoring the filter without any message. wdyt @johnaohara @willr3 ?

Reply:
With https://github.com/Hyperfoil/Horreum/pull/1977/commits/5f2afbb3c450fb8d3152227db959b65d1d7f50f6 I added a new test to ensure the return code is 500 when providing such invalid jsonpath.

Reply:
> In my implementation, it will throw an error which I think is much better than ignoring the filter without any message. wdyt @johnaohara @willr3 ?

We have seen this with grafana dashboards when a query was being built, every keystroke generated a request to Horreum, and if the filter was malformed (which is normally is when you are sending each keystroke while developing) it returned *all* the labelvalues.

IIRC @willr3 we talked and agreed that we would return an error or nothing if the filter was malformed.  I think the outstanding question was how grafana handled any errors if we returned an error

Reply:
>I think the outstanding question was how grafana handled any errors if we returned an error

That's a good point, @willr3 do you know already how Grafana handles errors? Otherwise I can try to investigate

Reply:
Ok I tried using the Grafana `Infinity` plugin and the panel will report the error as follows:
![Screenshot from 2024-10-03 09-24-18](https://github.com/user-attachments/assets/0b089403-ad8f-4578-8046-95681bd4114a)
![image](https://github.com/user-attachments/assets/fb0c46ba-8d7e-4e09-a265-487452866d62)

So I think we should be fine

Reply:
is it possible to provide a meaningful message to the user? or change the response in this case, I don't think it is an internal server error, but a malformed request i.e. HTTP 400

Reply:
>is it possible to provide a meaningful message to the user? or change the response in this case, I don't think it is an internal server error, but a malformed request i.e. HTTP 400

I need to check, I suspect all raised exceptions are handled as 500 for `labelValues` endpoint (e.g., even if the test is not found it returns 500 I guess - which is the case I tested so far in Grafana).. let me see if I can fix it. I agree that malformed request is much better and much more meaningful for the user.

Reply:
Changed as follows:
```
$ curl -s -X GET 'http://localhost:8080/api/test/109/labelValues' --url-query filter="$..Performance" -w "\n%{http_code}"
syntax error at or near "." of jsonpath input
400% 
```
```
[10:04](https://redhat-internal.slack.com/archives/D03URV8NQ6S/p1727942672984749)
$ curl -s -X GET 'http://localhost:8080/api/test/109111/labelValues' -w "\n%{http_code}"              [10:04:08]
Cannot find test 109111
404%
```

Reply:
That looks better!
don't mean to be a pita, but how is that rendered in grafana?

Reply:
it is rendered like this:
![image](https://github.com/user-attachments/assets/1d124566-c23e-41a5-9fe3-2dee798429ba)

I couldn't find any information about the error msg (I am using old Grafana so not sure if something changed in newer versions) but I think that it is already much more meaningful than before.

And the results is `null` (but this strictly depends on how you configured the panel I think):
![image](https://github.com/user-attachments/assets/f7b0815e-73fb-44da-adf7-9fc40ac7817c)